src/zenml/zen_server/routers/users_endpoints.py

```diff
@@ -308,7 +321,11 @@ def deactivate_user(
         """
         user = zen_store().get_user(user_name_or_id)
         if user.id != auth_context.user.id:
-            verify_permission_for_model(user, action=Action.UPDATE)
+            verify_permission_for_model(
+                user,
+                action=Action.UPDATE,
+                admin_status=auth_context.user.is_admin,
+            )
```

Comment:
Same thing here: a regular user should not be allowed to deactivate its own account (maybe we can even generalize that and say no account should be allowed to deactivate itself).

Reply:
No user can deactivate themselve from now on